fhir-server/src/main/java/com/ibm/fhir/server/operation/spi/FHIRResourceHelpers.java

```diff
@@ -31,6 +39,33 @@
     // Constant for indicating whether an update can be skipped when the requested update resource matches the existing one
     public static final boolean SKIPPABLE_UPDATE = true;
 
+    /**
+     * Validate an interaction for a specified resource type.
+     *
+     * @param interaction
+     *            the interaction to be performed
+     * @param resourceType
+     *            the resource type against which the interaction is to be performed
+     * @throws FHIROperationException
+     */
+    void validateInteraction(Interaction interaction, String resourceType) throws FHIROperationException;
+
+    /**
+     * Performs the heavy lifting associated with a 'create' interaction.
+     *
+     * @param type
+     *            the resource type specified as part of the request URL
+     * @param resource
+     *            the Resource to be stored.
+     * @param ifNoneExist
+     *            whether to create the resource if none exists
+     * @param doValidation
+     *            if true, validate the resource; if false, assume the resource has already been validated
+     * @return a FHIRRestOperationResponse object containing the results of the operation
+     * @throws Exception
+     */
+    FHIRRestOperationResponse doCreate(String type, Resource resource, String ifNoneExist, boolean doValidation) throws Exception;
```

Comment:
FHIRRestOperationResponse is this an Operation Response or an Interaction Response>?